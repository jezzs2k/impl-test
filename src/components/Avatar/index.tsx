import React from 'react';
import {Image, TouchableOpacity, ViewStyle, StyleProp} from 'react-native';

import {widthPercentageToDP} from '../../utils';

export const LINK_IMAGE_DEFAULT =
  'https://interactive-examples.mdn.mozilla.net/media/cc0-images/grapefruit-slice-332-332.jpg';

type AvatarPropType = {
  uri?: string;
  size?: number;
  styleAvatar?: StyleProp<ViewStyle>;
  borderWidth?: number;
  onPress?: () => void;
  activeOpacity?: number;
  borderColor?: string;
};

export const Avatar = ({
  uri,
  size,
  styleAvatar,
  borderWidth,
  onPress,
  activeOpacity,
  borderColor,
}: AvatarPropType) => (
  <TouchableOpacity
    style={[styleAvatar]}
    onPress={onPress}
    accessibilityRole={'button'}
    activeOpacity={activeOpacity}>
    <Image
      source={{uri}}
      style={{
        width: size,
        height: size,
        borderRadius: size / 2,
        borderWidth: borderWidth,
        borderColor: borderColor,
      }}
    />
  </TouchableOpacity>
);

Avatar.defaultProps = {
  uri: LINK_IMAGE_DEFAULT,
  size: widthPercentageToDP(20),
  styleAvatar: {},
  borderWidth: 0,
  onPress: () => {},
  activeOpacity: 1,
  borderColor: '#fff',
};
