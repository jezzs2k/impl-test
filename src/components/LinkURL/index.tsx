import React, {useCallback} from 'react';
import {
  Linking,
  StyleProp,
  Text,
  TouchableOpacity,
  TextStyle,
  StyleSheet,
} from 'react-native';
import {responsiveFontSetting} from '../../utils';

type LinkURLProptype = {
  url: string;
  styleUrl?: StyleProp<TextStyle>;
  title?: string;
};

export const LinkURL = ({
  url,
  styleUrl = {},
  title = 'Brower',
}: LinkURLProptype) => {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url);

    if (supported) {
      await Linking.openURL(url);
    }
  }, [url]);

  return (
    <TouchableOpacity onPress={handlePress}>
      <Text style={[styles.textSmall, styles.textBold, styles.link, styleUrl]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
  },
  link: {
    fontStyle: 'italic',
    textTransform: 'capitalize',
    fontSize: responsiveFontSetting(2),
    color: '#1068bf',
    textDecorationLine: 'underline',
  },
  textSmall: {
    fontSize: responsiveFontSetting(1.6),
    fontStyle: 'normal',
  },
});
