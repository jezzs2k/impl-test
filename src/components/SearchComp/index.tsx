import React from 'react';
import {View, TextInput, Text, StyleSheet} from 'react-native';
import {
  heightPercentageToDP,
  responsiveFontSetting,
  widthPercentageToDP,
} from '../../utils';

type SearchCompPropType = {
  onchangeText: (text: string) => void;
  textTitle: string;
  placeholder?: string;
};

export const SearchComp = ({
  onchangeText,
  textTitle,
  placeholder,
}: SearchCompPropType) => {
  return (
    <View style={styles.container}>
      <View style={styles.containerItem}>
        <Text style={styles.textBold}>{textTitle}</Text>
      </View>
      <TextInput
        style={styles.textInputStyle}
        placeholder={placeholder ?? 'Search user ...'}
        onChangeText={onchangeText}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    paddingHorizontal: widthPercentageToDP(5),
    paddingVertical: heightPercentageToDP(1),
  },
  containerItem: {
    marginBottom: heightPercentageToDP(1),
    paddingTop: heightPercentageToDP(1),
  },
  textStyle: {
    fontSize: responsiveFontSetting(2),
  },
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
    textTransform: 'capitalize',
  },
  textInputStyle: {
    paddingVertical: heightPercentageToDP(1),
    paddingHorizontal: widthPercentageToDP(3),
    borderWidth: 1,
    fontSize: responsiveFontSetting(2),
    borderRadius: 8,
    borderColor: '#bdc3c7',
  },
});
