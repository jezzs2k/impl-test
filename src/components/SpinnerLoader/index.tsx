import React from 'react';
import {View, StyleSheet} from 'react-native';
import Svg, {Rect} from 'react-native-svg';
import ContentLoader from 'react-native-masked-loader';

const getMaskedElement = () => {
  return (
    <Svg height={'250px'} width="100%" fill={'black'}>
      <Rect x="0" y="0" rx="8" ry="8" width="50%" height="16" />
      <Rect x="0" y="30" rx="9" ry="9" width="100%" height="128" />
      <Rect x="0" y="172" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="188" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="204" rx="4" ry="4" width="100%" height="8" />
    </Svg>
  );
};

export const BOSLoader = () => {
  const MaskedElement = getMaskedElement();
  return (
    <View style={styles.container}>
      <ContentLoader MaskedElement={MaskedElement} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    height: 285,
  },
});
