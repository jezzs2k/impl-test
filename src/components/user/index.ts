export * from './userDetailHeader';
export * from './userItem';
