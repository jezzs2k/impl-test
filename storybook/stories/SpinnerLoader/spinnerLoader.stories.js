import {storiesOf} from '@storybook/react-native';
import React from 'react';
import WrapperConent from '../WrapperConent';
import {BOSLoader} from '../../../src/components/SpinnerLoader';

storiesOf('Loading', module)
  .addDecorator(getStory => <WrapperConent>{getStory()}</WrapperConent>)
  .add('Default Loading', () => <BOSLoader />);
