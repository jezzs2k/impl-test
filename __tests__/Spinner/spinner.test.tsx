import React from 'react';
import renrender from 'react-test-renderer';

import {BOSLoader} from '../../src/components/SpinnerLoader';

jest.mock('react-native-masked-loader', () => {});

describe('SpinnerLoader', () => {
  it('Snapshot test SpinnerLoader component', () => {
    const component = renrender.create(<BOSLoader />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
