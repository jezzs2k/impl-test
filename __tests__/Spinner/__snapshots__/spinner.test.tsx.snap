// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerLoader Snapshot test SpinnerLoader component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": 285,
    }
  }
>
  <RNSVGSvgView
    bbHeight="250px"
    bbWidth="100%"
    fill="black"
    focusable={false}
    height="250px"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 250,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGRect
        height="16"
        rx="8"
        ry="8"
        width="50%"
        x="0"
        y="0"
      />
      <RNSVGRect
        height="128"
        rx="9"
        ry="9"
        width="100%"
        x="0"
        y="30"
      />
      <RNSVGRect
        height="8"
        rx="4"
        ry="4"
        width="100%"
        x="0"
        y="172"
      />
      <RNSVGRect
        height="8"
        rx="4"
        ry="4"
        width="100%"
        x="0"
        y="188"
      />
      <RNSVGRect
        height="8"
        rx="4"
        ry="4"
        width="100%"
        x="0"
        y="204"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
